nikola/__main__.py

```diff
@@ -122,9 +121,23 @@ def main(args=None):
 
     sys.path.append('')
     try:
-        loader = importlib.machinery.SourceFileLoader("conf", conf_filename)
-        conf = loader.load_module()
-        config = conf.__dict__
+        @contextmanager
+        def add_to_path(p):
+            """Temporarily adds `p` to `sys.path`."""
+            old_path = sys.path
+            sys.path = sys.path[:]
+            sys.path.insert(0, p)
+            try:
+                yield
+            finally:
+                sys.path = old_path
+
+        with add_to_path(os.path.dirname(conf_filename)):
+            with open(conf_filename, "rb") as file:
+                module = imp.load_module(conf_filename, file, conf_filename, (None, "rb", imp.PY_SOURCE))
```

Comment:
For testing whether the filename is working in every case please refer to the `test_config_with_illegal_filename`-test.

Actually passing a `name` argument causes python to either load `sys.modules[name]`, if it exists, rather than the file. Otherwise `sys.modules[name]` is set to the module loaded from the file.

By setting the name correctly you can simply re-import the module using an expression like
```python
import {name}
```

In our case we don't want to load any module from `sys.modules` and we also don't want to re-import it using an `import`-statement which is why I choose a distinguishable name as `name`-argument and delete it afterwards using `del sys.modules[conf_filename]`.
That way I make sure that the correct module is loaded in every case.

Though I think I might rework this behaviour and backup and restore `sys.modules` in the `add_to_path`-function.

Reply:
I now replaced said `add_to_path`-method by one called `temp_import` which ensures that both the `sys.path` and the `sys.modules` are not altered.